optaplanner-examples/src/test/java/org/optaplanner/examples/curriculumcourse/optional/score/CurriculumCourseConstraintProviderTest.java

```diff
@@ -0,0 +1,152 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.curriculumcourse.optional.score;
+
+import org.junit.jupiter.api.Test;
+import org.optaplanner.examples.curriculumcourse.domain.Course;
+import org.optaplanner.examples.curriculumcourse.domain.CourseSchedule;
+import org.optaplanner.examples.curriculumcourse.domain.Curriculum;
+import org.optaplanner.examples.curriculumcourse.domain.Day;
+import org.optaplanner.examples.curriculumcourse.domain.Lecture;
+import org.optaplanner.examples.curriculumcourse.domain.Period;
+import org.optaplanner.examples.curriculumcourse.domain.Room;
+import org.optaplanner.examples.curriculumcourse.domain.Teacher;
+import org.optaplanner.examples.curriculumcourse.domain.Timeslot;
+import org.optaplanner.examples.curriculumcourse.domain.UnavailablePeriodPenalty;
+import org.optaplanner.examples.curriculumcourse.domain.solver.CourseConflict;
+import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+
+class CurriculumCourseConstraintProviderTest {
+
+    private static final Curriculum CURRICULUM_1 = new Curriculum(1, "Curriculum1");
+    private static final Curriculum CURRICULUM_2 = new Curriculum(2, "Curriculum2");
+    private static final Teacher TEACHER_1 = new Teacher(1, "Teacher1");
+    private static final Teacher TEACHER_2 = new Teacher(2, "Teacher2");
+    private static final Course COURSE_1 = new Course(1, "Course1", TEACHER_1, 10, 20, 3, CURRICULUM_1);
+    private static final Course COURSE_2 = new Course(2, "Course2", TEACHER_2, 10, 10, 2, CURRICULUM_2);
+    private static final Course COURSE_3 = new Course(3, "Course3", TEACHER_1, 10, 5, 1, CURRICULUM_2);
+    private static final Room ROOM_1 = new Room(1, "Room1", 10);
+    private static final Room ROOM_2 = new Room(2, "Room2", 20);
+    private static final Timeslot FIRST_TIMESLOT = new Timeslot(0);
+    private static final Timeslot SECOND_TIMESLOT = new Timeslot(1);
+    private static final Day MONDAY = new Day(0);
+    private static final Day TUESDAY = new Day(1);
+    private static final Period PERIOD_1_MONDAY = new Period(0, MONDAY, FIRST_TIMESLOT);
+    private static final Period PERIOD_2_MONDAY = new Period(1, MONDAY, SECOND_TIMESLOT);
+    private static final Period PERIOD_1_TUESDAY = new Period(2, TUESDAY, FIRST_TIMESLOT);
+
+    private final ConstraintVerifier<CurriculumCourseConstraintProvider, CourseSchedule> constraintVerifier =
+            ConstraintVerifier.build(new CurriculumCourseConstraintProvider(), CourseSchedule.class, Lecture.class);
+
+    @Test
+    void conflictingLecturesDifferentCourseInSamePeriod() {
```

Comment:
same here: test methods in other test classes have a public modifier. Either we remove them across our code base, or we keep them public consistently.